packages/shared/src/components/Settings.tsx

```diff
@@ -111,7 +115,7 @@ export default function Settings({
             name="hide-read"
             className="my-3"
             checked={showOnlyUnreadPosts}
-            onToggle={onShowOnlyUnreadPosts}
+            onToggle={() => onToggleForLoggedUnUsers(toggleShowOnlyUnreadPosts)}
```

Comment:
We could use it directly as: `{onToggleForLoggedUnUsers(toggleShowOnlyUnreadPosts)}`

Reply:
I'm not sure you're right. Because `onToggleForLoggedUnUsers` doesn't generate a function.
So if we call it directly, it will execute the function immediately on render and this is not what we want to achieve here. I hope I didn't miss anything

Reply:
Ah woeps totally missed that one 🤯